crates/bevy_app/src/plugin.rs

```diff
@@ -144,12 +144,26 @@ mod sealed {
     impl<P: Plugin> Plugins<PluginMarker> for P {
         #[track_caller]
         fn add_to_app(self, app: &mut App) {
+            #[expect(
+                clippy::allow_attributes,
+                reason = "This lint only triggers sometimes, based on the features enabled."
+            )]
+            #[allow(unused_variables, reason = "plugin_name is only used with std enabled")]
             if let Err(AppError::DuplicatePlugin { plugin_name }) =
                 app.add_boxed_plugin(Box::new(self))
             {
-                panic!(
-                    "Error adding plugin {plugin_name}: : plugin was already added in application"
-                )
+                // We cannot use `warn!` here because this may occur before the Bevy logger is initialized.
+                // eprintln! is used as a fallback here, although it's generally not recommended for libraries to use directly.
+                //
+                // However, both `dbg!` and `eprintln!` are only available with the `std` feature.
+                // This is not a critical error, so we've chosen to simply ignore it in `no_std` environments.
+                #[cfg(feature = "std")]
+                {
```

Comment:
I think changing behavior based on the number of fields / size of the typee is confusing / hard to document / hard to understand. Additionally, nothing is stopping a ZST plugin from behaving differently at different times, or feeding on data that might change.

I think we either need to embrace  "unique plugins registered more than once ignore the configuration from the second plugin and warns" _or_ "double-registering unique plugins aborts app execution".

I do think the warning makes sense, as plugins shouldn't need to coordinate / anticipate what other plugins might also register. Of course, the true fix is deferred plugin init, which would embrace the "only one instance, with shared config" approach.

Reply:
Imo that's fine in this case, since it's not behavior you have to actively think about. It's just an instance of one less panic in a specialized case.

Reply:
Why bother though? Asking the other plugin to not add a ZST plugin has exactly the same effect in practice as what is proposed here. Imo we have a sensible and safe way to automate ecosystem tedium upstream with a tiny PR.